### PR TITLE
Propagate cancellation of gRPC requests

### PR DESCRIPTION
**Pull request by njhill**
@dtrifiro added a partial change for this while ago which was reverted to maintain compatibility with older vLLM version.

### CODE REVIEW

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 1]
```diff
@@ -253,7 +254,7 @@ async def Generate(
                 prompt=req.text,
                 prompt_token_ids=input_ids,
             )
-            kwargs = {}
+            kwargs = adapter_kwargs
```
Comment (dtrifiro) [inline src/vllm_tgis_adapter/grpc/grpc_server.py:257]: I guess we can just get rid of this>

**Review by dtrifiro — APPROVED** [review]

File: src/vllm_tgis_adapter/grpc/grpc_server.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
     extract_trace_headers,
     log_tracing_disabled_warning,
 )
+from vllm.utils import iterate_with_cancellation
```
[the remark this answers is not in the document]
Comment (njhill) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:29]: Missing from where? it's in v0.5.5...

[the remark this answers is not in the document]
Comment (dtrifiro) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:29]: You're right, I'm sorry, I thought it was in the adapter utils.

File: src/vllm_tgis_adapter/grpc/grpc_server.py (hunk 1 above)
Comment (njhill) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:257]: You mean get rid of separate `adapter_kwargs` var?

Comment (dtrifiro) [reply src/vllm_tgis_adapter/grpc/grpc_server.py:257]: yeah

**Pull request merged**